solidity/test/staker-rewards/MerkleDistributorTest.js

```diff
@@ -0,0 +1,210 @@
+
+        await rewardsDistributor.claim(
+          merkleRoot,
+          index,
+          account,
+          amount,
+          proof
+        )
+      }
+
+      const actualBalance = await keepToken.balanceOf(
+        rewardsDistributor.address
+      )
+
+      expect(actualBalance).to.eq.BN(
+        initialBalance.sub(claimedAmounts),
+        "invalid unbonded value"
+      )
+    })
+
+    it("should revert claiming transaction when proof is not valid", async () => {
+      const merkleRoot = merkle0.merkleRoot
+      const index = merkle0.claims[0].index
+      const account = merkle0.claims[0].account
+      const amount = merkle0.claims[0].amount
+      const proof = [
+        "0x1111111111111111111111111111111111111111111111111111111111111111",
+        "0xb335096692ef570690f2d858f2d52c268728d60b12a2a856f2691155ccf36377",
+      ]
+
+      await expectRevert(
+        rewardsDistributor.claim(merkleRoot, index, account, amount, proof),
+        "Invalid proof"
+      )
+    })
+
+    it("should successfuly claim rewards from multiple merkle roots", async () => {
+      const initialBalance = await keepToken.balanceOf(
+        rewardsDistributor.address
+      )
+
+      let claimedAmounts = new BN(0)
+
+      let merkleRoot = merkle0.merkleRoot
+      let index = merkle0.claims[0].index
+      let account = merkle0.claims[0].account
+      let amount = merkle0.claims[0].amount
+      let proof = merkle0.claims[0].proof
+
+      let value = new BN(amount)
+      claimedAmounts = claimedAmounts.add(value)
+
+      await rewardsDistributor.claim(merkleRoot, index, account, amount, proof)
+
+      merkleRoot = merkle1.merkleRoot
+      index = merkle1.claims[1].index
+      account = merkle1.claims[1].account
+      amount = merkle1.claims[1].amount
+      proof = merkle1.claims[1].proof
+
+      value = new BN(amount)
+      claimedAmounts = claimedAmounts.add(value)
+
+      await rewardsDistributor.claim(merkleRoot, index, account, amount, proof)
+
+      const actualBalance = await keepToken.balanceOf(
+        rewardsDistributor.address
+      )
+
+      expect(actualBalance).to.eq.BN(
+        initialBalance.sub(claimedAmounts),
+        "invalid unbonded value"
+      )
+    })
+
+    it("should revert when claiming a reward twice", async () => {
+      const merkleRoot = merkle0.merkleRoot
+      const index = merkle0.claims[0].index
+      const account = merkle0.claims[0].account
+      const amount = merkle0.claims[0].amount
+      const proof = merkle0.claims[0].proof
+
+      await rewardsDistributor.claim(merkleRoot, index, account, amount, proof)
+
+      await expectRevert(
+        rewardsDistributor.claim(merkleRoot, index, account, amount, proof),
+        "Reward already claimed"
+      )
+    })
+  })
+
+  async function allocateTokens(merkleRoot, amount) {
+    await keepToken.approve(rewardsDistributor.address, amount)
+    const allocated = await rewardsDistributor.allocate(merkleRoot, amount)
```

Comment:
Can we rename `allocated` to `receipt`? This is a pattern we use in other places in the code. `allocated` suggests we are getting a number equal to the number of tokens allocated while we are actually getting TX receipt.

Reply:
Done.